docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+      - "data/test/h5/paired/train1"
+      - "data/test/h5/paired/train2" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+```
+
+### Format key - Required
+
+The type of data we supply the data loaders will influence the behaviour, so we must
+specify the data type using the `format` key. Currently, DeepReg data loaders support
+nifti and h5 file types - alternate file formats will raise errors in the data loaders.
+To indicate which format to use, pass a string to this field as either "nifti" or "h5":
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+```
+
+Depending on the data type, DeepReg will expect the images and labels to be stored in
+specific structures: check the [data loader configuration](dataset_loader.html) for more
+details.
+
+### Labeled key - Required
+
+The `labeled` key indicates whether labels should be used during training. A Boolean is
+used to indicate the usage of labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  labeled: true
+```
+
+### Type key - Required
+
+The type of data loader used will depend on how one wants to train the network.
+Currently, DeepReg data loaders support the "paired", "unpaired" and "grouped" training
+strategies. Passing a string that doesn't match any of the above would raise an error.
+The data loader type would be specified using the `type` key:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
+    valid: "data/test/h5/paired/valid" # folder contains validation data
+    test: "data/test/h5/paired/test" # folder contains test data
+  format: "nifti"
+  type: "paired" # one of "paired", "unpaired" or "grouped"
+```
+
+#### Data loader dependent keys
+
+Depending on which string is passed to the `type` key, DeepReg will initialise a
+different data loader instance with different sampling strategies. These are reviewed in
+depth in the [dataset loader configuration](dataset_loader.html) documentation. Here we
+outline the arguments necessary to configure the different dataloaders.
+
+###### Sample_label - Required
+
+In the case that we have more than one label per image, we need to indicate to the
+loader which one to use. We can use the `sample_label` argument to indicate which method
+to use. Pass one of "sample", for random sampling, or "all" to use all the available
+labels:
+
+```yaml
+dataset:
+  dir:
+    train: "data/test/h5/paired/train1" # folders contains training data
```

Comment:
typo? the `train1` you got 1 here

Btw, it is one folder

Reply:
Oh, this was on purpose to specifically highlight using just one folder.